### PR TITLE
 Decode basic auth strings as UTF-8

### PR DESCRIPTION
**Pull request by jasonaowen**
SpotBugs noted that when we decoded the HTTP basic authentication header, we were doing so with the default encoding:

> [Reliance on default encoding](https://spotbugs.readthedocs.io/en/latest/bugDescriptions.html#dm-reliance-on-default-encoding-dm-default-encoding)
> 
> Found a call to a method which will perform a `byte` to `String` (or `String` to `byte`) conversion, and will assume that the default platform encoding is suitable. This will cause the application behaviour to vary between platforms. Use an alternative API and specify a charset name or Charset object explicitly.

According to RFC 7617, the client must encode in a character set that is
either compatible with ASCII (as UTF-8 is), or in UTF-8. (Per [this StackOverflow answer](https://stackoverflow.com/a/7243567/25637).)

Specify in our challenge header that we accept UTF-8, and decode the username and password as UTF-8 strings.

Add unit tests along the way, and do some refactoring in support of making this change.

I tested this by running the new unit tests, as well as by deploying and logging in through the API.

Issue #915 Use static analysis to find bugs

### CODE REVIEW

File: psm-app/cms-web/src/test/groovy/gov/medicaid/api/interceptors/BasicSecurityInterceptorTest.groovy [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    BasicSecurityInterceptor interceptor
+
+    def setup() {
+        interceptor = new BasicSecurityInterceptor(null)
+    }
+
+    @Unroll
+    def "username lookup throws on #bad header"(String bad, String header) {
+        when:
+        interceptor.getUsernameFromBasicAuthHeader(null)
```
Comment (PaulMorris) [inline psm-app/cms-web/src/test/groovy/gov/medicaid/api/interceptors/BasicSecurityInterceptorTest.groovy]: I'm probably missing something, but should this argument be `null`?  Seems like the `bad` and `header` arguments aren't used anywhere.  Unless there's something implicit going on.

Comment (jasonaowen) [reply psm-app/cms-web/src/test/groovy/gov/medicaid/api/interceptors/BasicSecurityInterceptorTest.groovy]: > I'm probably missing something, but should this argument be null? Seems like the ... `header` argument [isn't] used anywhere.

Oh my gosh, you're right! I initially had the `null` test separate, and then I combined it with other individual tests... poorly. Thank you for catching this! I will fix.

>  Seems like the `bad` ... argument [isn't] used anywhere

The `bad` variable is used in the test name `username lookup throws on #bad header`; you should see the various values substituted in the list of tests run.

**Comment by jasonaowen** [comment]
Thank you for the excellent review, @PaulMorris! I fixed the issue you found, and the tests are re-running now.

**Comment by jasonaowen** [comment]
Jenkins, test this please.

File: psm-app/cms-web/src/test/groovy/gov/medicaid/api/interceptors/BasicSecurityInterceptorTest.groovy (hunk 1 above)
Comment (PaulMorris) [reply psm-app/cms-web/src/test/groovy/gov/medicaid/api/interceptors/BasicSecurityInterceptorTest.groovy]: @jasonaowen Ah, got it, thanks for the explanation!

**Pull request merged**